src/streamlink/plugins/ltv_lsm_lv.py

```diff
@@ -1,45 +1,83 @@
 import logging
 import re
-from urllib.parse import urlparse
 
 from streamlink.plugin import Plugin, pluginmatcher
-from streamlink.plugin.api.utils import itertags
+from streamlink.plugin.api import validate
 from streamlink.stream.hls import HLSStream
 
 log = logging.getLogger(__name__)
 
 
 @pluginmatcher(re.compile(
-    r"https?://ltv\.lsm\.lv/lv/tieshraide"
+    r"https://ltv\.lsm\.lv/lv/tiesraide"
 ))
 class LtvLsmLv(Plugin):
     """
     Support for Latvian live channels streams on ltv.lsm.lv
     """
+
+    API_URL = "https://player.cloudycdn.services/player/ltvlive/channel/{channel_id}/"
+    _json_data_re = re.compile(r'teliaPlayer\((\{.*?\})\);', re.DOTALL)
+
+    main_page_schema = validate.Schema(
+        validate.parse_html(),
+        validate.xml_xpath_string(".//iframe[contains(@src, 'ltv.lsm.lv/embed')][1]/@src"),
+        validate.url()
+    )
+
+    embed_code_schema = validate.Schema(
+        validate.parse_html(),
+        validate.xml_xpath_string(".//live/@*"),
+        validate.parse_json(),
+        {"source": {"embed_code": validate.text}},
```

Comment:
`validate.text` should be replaced with `str` since the removal of py2. Same down below...

Reply:
I disagree with this suggestion, I don't think `str` is a good name as it can be conflated with the actual type. Also, every other instance (that I could see) is `validate.text`.

Reply:
Well, it's no big deal, but the reason why I suggested the change is that `validate.text` is just an alias for the actual `str` type and it executes the same singledispatch validation method `validate_type` as other base types like `int`. `text` a left-over module export of the py2 compatibility era, and removing it would've required changing all validation schemas at that time, which didn't make much sense, so it was left.
- https://github.com/streamlink/streamlink/commit/c94a8107916a284e69c8c14dda3894d9f8d81495#diff-c415fd588e79ab1a5ef3a0fc4fbf7a90769778435b75a80b589e9c7b02afcbd7R37
- https://github.com/streamlink/streamlink/blob/37ee5347508980ee768ac0695714f01c54600933/src/streamlink/plugin/api/validate.py#L427-L436

All newly added or edited plugins are using the base types in their validation schemas.
```sh
$ grep ': str' src/streamlink/plugins/*.py | wc -l
168
$ grep 'validate.text' src/streamlink/plugins/*.py | wc -l
149
```

Reply:
fair enough :)